torchgen/model.py

```diff
@@ -571,11 +575,15 @@ def from_yaml(
                 cpp.name(func), structured=False
             )
 
-        assert not (
-            DispatchKey.CompositeExplicitAutograd in dispatch
-            and DispatchKey.CompositeImplicitAutograd in dispatch
-        ), (
-            "cannot specify both CompositeExplicitAutograd and CompositeImplicitAutograd on a single kernel; each "
+        composites_in_dispatch = [
+            d for d in dispatch if
+            d == DispatchKey.CompositeExplicitAutograd
+            or d == DispatchKey.CompositeExplicitAutogradWithMutations
+            or d == DispatchKey.CompositeImplicitAutograd]
+
+        assert len(composites_in_dispatch) <= 1, (
+            "cannot specify more than one of CompositeExplicitAutograd, CompositeExplicitAutogradWithMutations, "
+            "or CompositeImplicitAutograd on a single kernel; each "
```

Comment:
I don't think this is right. I could have both CompositeExplicitAutograd and CompositeExplicitAutogradWithMutations, and under the precedence you have currently implemented this would not be a no-op because CompositeExplicitAutogradWithMutations would be applied to all the non-functional backends and CompositeExplicitAutograd would be applied to the functional backends. This might even be useful if you prefer the mutating composite in eager mode.

Reply:
Oh, that's a good point (although I'm planning on renaming this key to something like `CompositeExplicitAutogradWithAliasing`, so the mutation bit may not apply). Ok, it seems reasonable to allow for an operator to provide kernels for both keys.